ngraph_bridge/grappler/ngraph_optimizer.cc

```diff
@@ -272,12 +273,21 @@ Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
                "Graph with Variables Rewritten for Tracking");
   }
 
+  AddSessionNameAttr(
```

Comment:
Add a utility function is_tensorboard_diagnostics_requested and call these functions only if that is true.

You can check for the environment flag in that utility

Reply:
already have an environment flag for tensorboard (NGRAPH_TF_TB_LOGDIR). i'll check if this exists, and if so, do AddSessionNameAttr

Reply:
yes, the check for the env flag can be made into a utility function, as you will have to do it many times